src/main/java/io/confluent/connect/jdbc/sink/JdbcDbWriter.java

```diff
@@ -79,12 +80,15 @@ void closeQuietly() {
     cachedConnectionProvider.close();
   }
 
-  TableId destinationTable(String topic) {
-    final String tableName = config.tableNameFormat.replace("${topic}", topic);
+  TableId destinationTable(SinkRecord record) {
+    String schemaNameKey = config.schemaNameFormat;
+    String schemaName = (String) ((Struct)record.value()).get(schemaNameKey);
```

Comment:
Yes, this looks good. To use `projectId` though, we'd need to select something from `record.key()`, right?